Firmware/ultralcd.cpp

```diff
@@ -5228,6 +5228,10 @@ static void lcd_main_menu()
         MENU_ITEM_FUNCTION_P(PSTR("tst - Restore"), lcd_menu_test_restore);
 #endif //RESUME_DEBUG
 
+#ifdef DEBUG_PRINTER_STATES
+    debug_printer_states();
```

Comment:
I have a bad feeling about this. `lcd_main_menu()` is called very frequently and dumping this much stuff on the serial stream will likely do more harm than good. Does the printer actually keep up with this much serial prints?

Reply:
Another point is, what happen when the printer is not rendering the main menu? Then these debug prints are ignored.

Reply:
You are correct, this menu will dump lot of information over serial and should only be used with the MK404 or with a subset of the messages.
I added this to be able to troubleshoot some LCD menu's.